### PR TITLE
feat(er): add a new datasource to query the list of flow log

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:
Add a new datasource to query the list of flow log
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. Add a new datasource.
1.1 the datasource is used to query the list of flow log.
1.2 add the datasource documentation (docs/datasource/er_flow_logs).
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
root@ecs-ruwenqiang:/home/huawei/go/src/github.com/huaweicloud/terraform-provider-huaweicloud (br_flowlogs)$ make testacc TEST="./huaweicloud/services/acceptance/er" TESTARGS="-run TestAccDatasourceFlowLogs_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/er -v -run TestAccDatasourceFlowLogs_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceFlowLogs_basic
=== PAUSE TestAccDatasourceFlowLogs_basic
=== CONT  TestAccDatasourceFlowLogs_basic
--- PASS: TestAccDatasourceFlowLogs_basic (197.01s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/er        197.050s
```

### CODE REVIEW

File: docs/data-sources/er_flow_logs.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+Use this data source to get the list of flow logs.
+
+## Example Usage
+
+```hcl
+variable "resource_id" {}
+
+data "huaweicloud_er_flow_logs" "test" {
+  resource_id = var.resource_id
```
Comment (ShiChangkuo) [inline docs/data-sources/er_flow_logs.md:17]: the required `instance_id` param is missing.

File: huaweicloud/services/er/data_source_huaweicloud_er_flow_logs.go [hunk 2]
```diff
@@ -0,0 +1,279 @@
+// ---------------------------------------------------------------
+// *** AUTO GENERATED CODE ***
+// @Product ER
+// ---------------------------------------------------------------
+
```
Comment (ShiChangkuo) [inline huaweicloud/services/er/data_source_huaweicloud_er_flow_logs.go]: those lines should be deleted

File: huaweicloud/services/er/data_source_huaweicloud_er_flow_logs.go [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+	listFlowLogsQueryParams := buildListFlowLogsQueryParams(d, cfg)
+	listFlowLogsPath += listFlowLogsQueryParams
+
+	listFlowLogsOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	listFlowLogsResp, err := listFlowLogsClient.Request("GET", listFlowLogsPath, &listFlowLogsOpt)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving flow logs")
```
Comment (ShiChangkuo) [inline huaweicloud/services/er/data_source_huaweicloud_er_flow_logs.go]: directly return the error will be fine
```suggestion
		return diag.Errorf("error retrieving flow logs: %s", err)
```

File: huaweicloud/services/acceptance/er/data_source_huaweicloud_er_flow_logs_test.go [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+  auto_create_vpc_routes = true
+
+  tags = {
+    foo = "bar"
+  }
+}
+`, common.TestVpc(name), name, bgpAsNum)
+}
+
+func testAccFlowLogsDataSource_base_2(name string, bgpAsNum int) string {
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/er/data_source_huaweicloud_er_flow_logs_test.go]: you can merge `testAccFlowLogsDataSource_base_1` and `testAccFlowLogsDataSource_base_2` into `testAccFlowLogsDataSource_base`

File: docs/data-sources/er_flow_logs.md (hunk 1 above)
Comment (ruwenqiang123) [reply docs/data-sources/er_flow_logs.md:17]: done

File: huaweicloud/services/er/data_source_huaweicloud_er_flow_logs.go (hunk 2 above)
Comment (ruwenqiang123) [reply huaweicloud/services/er/data_source_huaweicloud_er_flow_logs.go]: done

File: huaweicloud/services/er/data_source_huaweicloud_er_flow_logs.go (hunk 3 above)
Comment (ruwenqiang123) [reply huaweicloud/services/er/data_source_huaweicloud_er_flow_logs.go]: done

File: huaweicloud/services/acceptance/er/data_source_huaweicloud_er_flow_logs_test.go (hunk 4 above)
Comment (ruwenqiang123) [reply huaweicloud/services/acceptance/er/data_source_huaweicloud_er_flow_logs_test.go]: done

**Pull request merged**